dags/bqetl_unified.py

```diff
@@ -101,6 +102,13 @@
     with TaskGroup(
         "telemetry_derived__unified_metrics__v1_external"
     ) as telemetry_derived__unified_metrics__v1_external:
+        ExternalTaskMarker(
+            task_id="bqetl_analytics_aggregations__wait_for_telemetry_derived__unified_metrics__v1",
+            external_dag_id="bqetl_analytics_aggregations",
+            external_task_id="wait_for_telemetry_derived__unified_metrics__v1",
+            execution_date="{{ (execution_date - macros.timedelta(seconds=7200)).isoformat() }}",
+        )
```

Comment:
I'm unclear why this `ExternalTaskMarker` (which looks correct) is only now getting generated, as I don't think the changes I made should make a difference with that.

Reply:
mmh that is odd...

Reply:
I managed to figure this out:  it's due to [using `TaskRef` objects as dictionary keys in `DagCollection.downstream_tasks`](https://github.com/mozilla/bigquery-etl/blob/332c569bed4be3c3896c8cd6daa0d40ed7acc83f/bigquery_etl/query_scheduling/dag_collection.py#L37-L40) and ending up with a different `TaskRef` after the task has its `date_partition_offset` set (and that was exacerbated by the [upstream dependencies being calculated in parallel](https://github.com/mozilla/bigquery-etl/blob/332c569bed4be3c3896c8cd6daa0d40ed7acc83f/bigquery_etl/query_scheduling/dag_collection.py#L161-L162)).

Also, while doing test runs with/without [parallelization of calculating upstream & downstream dependencies](https://github.com/mozilla/bigquery-etl/blob/332c569bed4be3c3896c8cd6daa0d40ed7acc83f/bigquery_etl/query_scheduling/dag_collection.py#L161-L165) I noticed without parallelization is actually a bit faster and lets you see the log messages.

I'm going to make some changes to address what I found and re-request review.